Fix error on iOS Payments page

luacmartins opened the pull request:
### Details
Changes:
1. Removes `ScrollView`
2. Passes `ListHeaderComponent` prop to `PaymentMethodList`

### Fixed Issues
$ https://github.com/Expensify/App/issues/7087

### Tests
1. Launch the iOS app
2. Navigate to `Settings > Payments`
3. Verify that no error is thrown and you can still see the account balance.

### QA Steps
Steps above.

### Tested On

- [x] Web
- [X] Mobile Web
- [x] Desktop
- [X] iOS
- [x] Android

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web

https://user-images.githubusercontent.com/22219519/148604724-dcf14af8-f573-479f-918e-b097b6804487.mov

#### Mobile Web

https://user-images.githubusercontent.com/22219519/148604643-85795d5f-723e-455e-a3d7-85a466960957.mp4

#### Desktop

https://user-images.githubusercontent.com/22219519/148605577-34b62bf0-37b1-4030-878e-e08f499ea2f1.mov

#### iOS

https://user-images.githubusercontent.com/22219519/148604517-25de1029-0f05-42f8-94c2-b420ff1d2a30.mp4

#### Android

https://user-images.githubusercontent.com/22219519/148604546-41fee23a-7600-4eea-9109-2026c681835a.mov

luacmartins committed:
pass ListHeaderComponent and remove ScrollView

luacmartins committed:
rm changes to podfile.lock

luacmartins committed:
add new line at end of file

luacmartins committed:
pass in function

stitesExpensify commented on src/pages/settings/Payments/PaymentsPage.js:
```diff
@@ -120,22 +119,20 @@ class PaymentsPage extends React.Component {
-                        />
-                    </ScrollView>
+                    <PaymentMethodList
+                        ListHeaderComponent={() => (
```
Just to confirm, this header is still sticky right? As in if we have a ton of accounts and the user scrolls it's not going to scroll too?

luacmartins replied:
Oh is the header (Account balance + Payments method text) supposed to be sticky? I thought it was supposed to scroll since it scrolls in the current implementation.

marcaaron replied:
+1 

Could we also just put the header outside of the `FlatList` if we are worried about this?

luacmartins replied:
Current behavior:
https://user-images.githubusercontent.com/22219519/148615425-d1e165b6-7e60-4841-a75f-e4254423bf1b.mov

luacmartins replied:
>Could we also just put the header outside of the FlatList if we are worried about this?
Yes, the solution is simpler if the header is supposed to be sticky. I just thought it had to scroll since that's how ti behaves now.

stitesExpensify replied:
Awesome! Thanks

luacmartins committed:
revert changes

luacmartins committed:
change ScrollView to View

luacmartins commented:
Updated! @stitesExpensify and I discussed 1:1 and the header should be sticky. I updated the solution to account for that. Here's the new behavior:

https://user-images.githubusercontent.com/22219519/148616770-58244bfe-0aa5-44b1-ab39-38148a3d67d7.mp4

stitesExpensify commented:
Looks great! I'll merge when e2e ends in what I assume will be 3 hours :lolsob:

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1678921547) to staging by @stitesExpensify in version: 1.1.26-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1678991530) to staging by @stitesExpensify in version: 1.1.27-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1685973556) to production by @roryabraham in version: 1.1.27-1 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
